tests/transforms/test_create_slapp_inputs.py

```diff
@@ -0,0 +1,146 @@
+import json
+import h5py
+import pytest
+from pathlib import Path
+
+import ophys_etl.transforms.create_slapp_inputs as csli
+
+
+def experiment(args, tmp_path):
+    eid = args.get("experiment_id")
+    edir = tmp_path / f"{eid}"
+    edir.mkdir()
+    bpath = edir / "binarize_output.json"
+    with open(bpath, "w") as f:
+        json.dump(args.get("rois"), f)
+    tpath = edir / "roi_traces.h5"
+    with h5py.File(tpath, "w") as f:
+        f.create_dataset("data", data=[])
+    mpath = edir / "movie.h5"
+    with h5py.File(mpath, "w") as f:
+        f.create_dataset("data", data=[])
+
+    experiment_dict = {
+            'experiment_id': eid,
+            'binarized_rois_path': str(bpath),
+            'traces_h5_path': str(tpath),
+            'movie_path': str(mpath)}
+    if 'local_ids' in args:
+        experiment_dict['local_ids'] = args.get('local_ids')
+
+    return experiment_dict
+
+
+@pytest.fixture
+def experiments_fixture(request, tmp_path):
+    experiments = list([experiment(i, tmp_path)
+                        for i in request.param.get('experiments')])
+    yield experiments
+
+
+@pytest.mark.parametrize("n_roi_total", [7, 8])
+@pytest.mark.parametrize(
+        "experiments_fixture",
+        [
+            {'experiments': [
+                {
+                    'experiment_id': 1001,
+                    'rois': [
+                        {'id': 0, 'exclusion_labels': []},
+                        {'id': 1, 'exclusion_labels': []},
+                        {'id': 2, 'exclusion_labels': ['not empty']},
+                        {'id': 3, 'exclusion_labels': []},
+                        {'id': 4, 'exclusion_labels': []}]},
+                {
+                    'experiment_id': 1002,
+                    'rois': [
+                        {'id': 0, 'exclusion_labels': ['not empty']},
+                        {'id': 1, 'exclusion_labels': []},
+                        {'id': 2, 'exclusion_labels': []},
+                        {'id': 3, 'exclusion_labels': []},
+                        {'id': 4, 'exclusion_labels': []}]},
+            ]},
+            ], indirect=['experiments_fixture'])
+def test_select_rois(experiments_fixture, n_roi_total):
+    print(json.dumps(experiments_fixture, indent=2))
+    selected = csli.select_rois(experiments_fixture, n_roi_total, 42)
+    n = 0
+    for exp in selected:
```

Comment:
The functional programmer in me wants:

`n = reduce(lambda x, exp: x + len(exp["local_ids"]), selected, 0)`

But you don't need to do that 👯

Reply:
I didn't make it functional, but I did one-line it with comprehension